src/CacheManager.SystemRuntimeCaching/MemoryCacheHandle`1.cs

```diff
@@ -35,21 +35,33 @@ public class MemoryCacheHandle<TCacheValue> : BaseCacheHandle<TCacheValue>
         /// <param name="configuration">The cache handle configuration.</param>
         /// <param name="loggerFactory">The logger factory.</param>
         public MemoryCacheHandle(ICacheManagerConfiguration managerConfiguration, CacheHandleConfiguration configuration, ILoggerFactory loggerFactory)
+            : this(managerConfiguration, configuration, loggerFactory, null)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MemoryCacheHandle{TCacheValue}"/> class.
+        /// </summary>
+        /// <param name="managerConfiguration">The manager configuration.</param>
+        /// <param name="configuration">The cache handle configuration.</param>
+        /// <param name="loggerFactory">The logger factory.</param>
+        /// <param name="memoryCacheOptions">The vendor specific options.</param>
+        public MemoryCacheHandle(ICacheManagerConfiguration managerConfiguration, CacheHandleConfiguration configuration, ILoggerFactory loggerFactory, RuntimeMemoryCacheOptions memoryCacheOptions)
             : base(managerConfiguration, configuration)
         {
             NotNull(configuration, nameof(configuration));
             NotNull(loggerFactory, nameof(loggerFactory));
-
             Logger = loggerFactory.CreateLogger(this);
             _cacheName = configuration.Name;
 
             if (_cacheName.ToUpper(CultureInfo.InvariantCulture).Equals(DefaultName.ToUpper(CultureInfo.InvariantCulture)))
             {
+                //we can't change default cache configuration by code, can we?
```

Comment:
Instead of silently discard the passed in options, this should probably throw a hard exception saying "Configuring the Default instance of MemoryCache via RuntimeMemoryCacheOptions is not supported, please specify a different cache handle name."

Or, alternatively to make it easier to use, create a named MemoryCache instance with a random name (Guid.NewGuid) and log an info message saying that a new named instance of MemoryCache was created for the given RuntimeMemoryCacheOptions because configuring the Default instance is not supported.

Reply:
I think I prefer to just throw a hard exception and avoid some masking of the problem.
Using a named instance could cause some confusion, for instance using the app/web.config to configure the default instance but then CacheManager would use a different instance. Or some other user code could access directly to the Default instance.

Reply:
yup agree